drivers/hmis/app/models/hmis/hud/household.rb

```diff
@@ -17,14 +17,16 @@ class Hmis::Hud::Household < Hmis::Hud::Base
   alias_attribute :household_id, :HouseholdID
 
   replace_scope :viewable_by, ->(user) do
-    viewable_households = joins(:enrollments).
-      merge(Hmis::Hud::Enrollment.viewable_by(user)). # does Data Source filter
-      pluck(:HouseholdID)
-
-    where(HouseholdID: viewable_households)
+    # correlated subquery
+    p_t = Hmis::Hud::Project.arel_table
+    hh_t = Hmis::Hud::Household.arel_table
+    cond = p_t[:ProjectID].eq(hh_t[:ProjectID]).and(p_t[:data_source_id].eq(hh_t[:data_source_id]))
+    projects = Hmis::Hud::Project.with_access(user, :can_view_enrollment_details, :can_view_project, mode: 'all').where(cond)
+    where(projects.arel.exists)
   end
 
   scope :client_matches_search_term, ->(text_search) do
+    # FIXME: the sort order from text search is not preserved
```

Comment:
Is this something we need to fix now? or more of a TODO?

Reply:
Definitely a to-do. Captured it in a ticket: https://www.pivotaltracker.com/story/show/186384617